support.go

```diff
@@ -0,0 +1,175 @@
+// The string will be the line from out.txt that calls the custom syntax, this is so any necessary parameters can be extracted out.
+// The string slice is the slice containing all of the lines output to stdout
+// The integer will be the current position in the above slice that the grader is in
+// Furthermore, the functions will return two integer values.
+// The first integer represents the result score, which will hold either 1 or 0.
+// The second integer will be the new position in the slice containing stdout
+func initSyntaxDictionary() map[string]func(string, []string, int) (string, int, []string) {
+	SyntaxDictionary := make(map[string]func(string, []string, int) (string, int, []string))
+
+	SyntaxDictionary["menu"] = func(menuCall string, StdOutput []string, startPos int) (string, int, []string) {
+		// Menu keyword comes in the form of !menu(x),
+		// where x is a digit representing the number of options the menu should display.
+		// Menus should resemble the following form (using x = 3):
+		/*
+			Title of the menu \n
+			1. Option \n
+			2. Option \n
+			3. Option \n
+			Prompt:
+		*/
+		// This shape can be simplified to following:
+		/*
+			string
+			string leading with 2
+			string leading with 1
+			string leading with 3
+			string ending with :
+		*/
+		// Declare return variables
+		var feedback string = ""
+		var newPos int
+		modifiedStdout := StdOutput
+
+		// First, extract the value of x CONVERT
+		x_value := extractXValue(menuCall)
+
+		if x_value < 1 || x_value > 9 {
+			return "Invalid menu parameter", startPos, StdOutput
+		}
+		// Get current position in stdout
+		curr := startPos
+
+		// Now, evaluate the menu shape.
+		// Check for title
+		if StdOutput[curr] == "" {
+			feedback = "No title" // Check failed
+		}
+
+		curr++
+
+		if feedback != "" {
+			var additionalFeedback string
+			curr, additionalFeedback = hasOptions(StdOutput, x_value, curr)
+			feedback += additionalFeedback
+		} else {
+			curr, feedback = hasOptions(StdOutput, x_value, curr)
+		}
+
+		// Check to see if curr exceeds bounds again
+		if curr > len(StdOutput)-1 {
+			feedback += "Position exceeds bounds"
+			curr-- // Move curr back if it exceeds the bounds of StdOutput
+			newPos = curr
+			return feedback, newPos, modifiedStdout
+		}
+
+		var promptPassed bool
+		promptPassed, modifiedStdout = hasPrompt(StdOutput, curr)
+
+		if !promptPassed {
+			feedback += " No prompt"
+			curr--
+		}
+
+		// Finally, update new position with current
+		newPos = curr
+
+		return feedback, newPos, modifiedStdout
+	}
+
+	SyntaxDictionary["ignore"] = func(ignoreCall string, StdOutput []string, startPos int) (string, int, []string) {
+		var feedback string = "Output ignored"
+		// Core functionality of ignore.... just skip da line
+
+		return feedback, startPos, StdOutput
+	}
+	return SyntaxDictionary
+}
+
+func extractXValue(s string) int {
+	x_index := strings.Index(s, "(") + 1
+	x_value := int(s[x_index]) - 48 // Ascii subtract
```

Comment:
may be worth changing this to something a little more legible in the future

perhaps use regex to capture between the parens, then use strconv.Atoi to convert the captured string to a number.

Shorter term, we should add input validation. What happens when someone inputs `!menu(z)`?

Reply:
I see error checking is handled outside. Maybe worth adding parameters to this function that can specify the min and max?